From 3e208944aeebf4b4d5529901f84bbfd88f75b1f7 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Wed, 14 Sep 2011 16:26:11 +0800
Subject: [PATCH] ENGR00156183-2 sii902x: add error handle of put_pins for probe fail

there is hardware pin conflict between sii902x DET and egalax touch screen
on mx53 loco board. Request gpio during sii902x probe can fix this conflict
only when these two module not co-exist.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/video/mxc/mxcfb_sii902x.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/video/mxc/mxcfb_sii902x.c b/drivers/video/mxc/mxcfb_sii902x.c
index 43613bb..ad23cd9 100644
--- a/drivers/video/mxc/mxcfb_sii902x.c
+++ b/drivers/video/mxc/mxcfb_sii902x.c
@@ -1204,6 +1204,8 @@ static int __devinit sii902x_probe(struct i2c_client *client,
 	return ret;
 
 init_failed:
+	if (plat->put_pins)
+		plat->put_pins();
 get_pins_failed:
 	platform_device_unregister(sii902x->pdev);
 register_pltdev_failed:
-- 
1.5.4.4

